tlsdialer.go

```diff
@@ -201,7 +201,14 @@ func (d *Dialer) DialForTimings(network, addr string) (*ConnWithTimings, error)
 	}
 	conn := tls.UClient(rawConn, configCopy, chid)
 	if d.ClientSessionState != nil {
-		conn.SetSessionState(d.ClientSessionState)
+		hasCachedSession := false
+		if configCopy.ClientSessionCache != nil {
+			sessionCacheKey := clientSessionCacheKey(result.ResolvedAddr, configCopy)
```

Comment:
Go uses `rawConn.GetRemoteAddr` for that first argument. Any reason not to do the same?

https://github.com/golang/go/blob/master/src/crypto/tls/handshake_client.go#L241

Reply:
Sure, that should work. Will switch.

Reply:
Okay, switched and verified that packet capture still looks good.